Providers/Profile.swift

```diff
@@ -677,34 +678,38 @@ open class BrowserProfile: Profile {
             notifySyncing(notification: NotificationProfileDidStartSyncing)
         }
 
-        fileprivate func endSyncing(_ result: Maybe<EngineResults>?) {
-            // loop through status's and fill sync state
+        fileprivate func endSyncing(_ result: SyncOperationResult) {
+            // loop through statuses and fill sync state
             syncLock.lock()
             defer { syncLock.unlock() }
             log.info("Ending all queued syncs.")
 
-            if let syncResult = result {
-                syncDisplayState = SyncStatusResolver(engineResults: syncResult).resolveResults()
+            syncDisplayState = SyncStatusResolver(engineResults: result.engineResults).resolveResults()
```

Comment:
There might be a subtle little bug here.

If we previously called `endSyncing` with `None`, we wouldn't change the display state.

Now we pass the none-`Maybe` to `SyncStatusResolver`, which resolves to `SyncDisplayState.Bad(message: nil)`.

Is this change intentional?